File: frontend/src/api/board.ts
import { authFetch } from './auth'

export type BoardProgress = {
  done: number
  total: number
}

export type BoardHabit = {
  id: string
  user_id: string
  category_id?: string | null
  name: string
  description?: string | null
  color?: string | null
  type: 'boolean' | 'numeric' | 'duration'
  frequency: string
  target_value?: number | null
  unit?: string | null
  sort_order: number
  is_completed: boolean
  streak: number
}

export type Board = {
  date: string
  is_editable: boolean
  progress: BoardProgress
  habits: BoardHabit[]
}

export async function getBoard(date: string): Promise<Board> {
  const res = await authFetch(`/api/v1/board/${date}`)
  if (!res.ok) {
    throw new Error(`Failed to fetch board: ${res.statusText}`)
  }
  return res.json() as Promise<Board>
}

File: backend/internal/domain/board.go
package domain

type BoardProgress struct {
	Done  int `json:"done"`
	Total int `json:"total"`
}

type BoardHabit struct {
	ID          string         `json:"id"`
	UserID      string         `json:"user_id"`
	CategoryID  *string        `json:"category_id,omitempty"`
	Name        string         `json:"name"`
	Description *string        `json:"description,omitempty"`
	Color       *string        `json:"color,omitempty"`
	Type        HabitType      `json:"type"`
	Frequency   HabitFrequency `json:"frequency"`
	TargetValue *float64       `json:"target_value,omitempty"`
	Unit        *string        `json:"unit,omitempty"`
	SortOrder   int            `json:"sort_order"`
	IsCompleted bool           `json:"is_completed"`
	Streak      int            `json:"streak"`
}

type Board struct {
	Date       string        `json:"date"`
	IsEditable bool          `json:"is_editable"`
	Progress   BoardProgress `json:"progress"`
	Habits     []BoardHabit  `json:"habits"`
}
